apps/explorer/src/hooks/useZodForm.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+
+import type { Resolver } from '@hookform/resolvers/zod';
+import type { UseFormProps } from 'react-hook-form';
+import type { z, ZodTypeAny } from 'zod';
+
+export function useZodForm<S extends ZodTypeAny, TContext = any>(
```

Comment:
I think we can get away from some of these types (mostly because I don't think we need options). I also think we can shove this into the top-level API:

```typescript
import type { ZodSchema, TypeOf } from 'zod';

interface UseZodFormProps<T extends ZodSchema<any>>
	extends UseFormProps<TypeOf<T>> {
	schema: T;
}

export const useZodForm = <T extends ZodSchema<any>>({
	schema,
	...formConfig
}: UseZodFormProps<T>) => {
	return useForm({
		...formConfig,
		resolver: zodResolver(schema),
	});
};
```